src/programs/match_template/match_template.cpp

```diff
@@ -381,6 +384,21 @@ bool MatchTemplateApp::DoCalculation()
 	wxString	result_output_filename = my_current_job.arguments[38].ReturnStringArgument();
 	float		min_peak_radius = my_current_job.arguments[39].ReturnFloatArgument();
 	bool		use_gpu   = my_current_job.arguments[40].ReturnBoolArgument();
+	int			max_threads =  my_current_job.arguments[41].ReturnIntegerArgument();
+
+	if (is_running_locally == false) max_threads = number_of_threads_requested_on_command_line; // OVERRIDE FOR THE GUI, AS IT HAS TO BE SET ON THE COMMAND LINE...
+
```

Comment:
Just following up - if i understand this correctly then if running through the command line without GPU it will not ask or use threads.  If running from the command line with GPU it will ask num threads and use that in the program.  If running through the GUI using the GPU it will use the num threads specified in the run profiles and if running through the GUI without GPU it will crash if the number of threads in the run profile is greater than 1?

Reply:
That is right, maybe a more graceful exit over "MyAssertTrue" would be a good idea? I guess we could also just optionally override and set max_threads=1 if !use_gpu, maybe printing a warning? I guess that would probably be better

Reply:
Right if you use SendInfo for the warning it will report it back to the gui.

Reply:
Okay, I changed this, but then screwed things up trying to rebase the two add on commits to make a cleaner history.  I think the easiest and cleanest fix for this particular situation is to 

1) Close this PR
2) Create a new branch (prob after you merge in PR 289)
3) Merge the changes from this PR, including the two fixes based on your suggestions into the new branch, that will have a single, non-divergent commit history and re-submit a new PR

I'll wait for a "thumbs up" from both @arohou and @timothygrant80 before proceeding.

Reply:
Can't you just do a commit now?     You are suggesting committing without the changes, then doing another PR for the changes?  I don't know git very well, but don't understand why that is necessary?

Reply:
Oh sorry - you just want to close this PR, then make a new PR with everything it?  I think that's fine if you'd rather do that - sounds like extra work for you though?

Reply:
Ben - your plan sounds good to me. I think it's worth the effort so that the history is cleaner for code archeologists of the future.

Reply:
I'll spare the explanation, but I was able to start over, and squash those two commits into the original, with the only change to this PR being the commit hash, which should be fine.

I'm not sure what I did would have worked if other people had been editing the files, since it required a force push to my downstream remote after re-basing, but in this case everything seems to have worked out.

Reply:
@timothygrant80 here is that second change in case you want to check it out before merging

https://github.com/bHimes/cisTEM_downstream_bah/blob/4fc6328f554dd3699d2d60b81191f04e391d3606/src/programs/match_template/match_template.cpp#L389-L401

Reply:
Not sure if it is an issue, but for me now the details section is blue and it says "This commit does not belong to any branch on this repository."

Reply:
Sorry, what is the "details section"?

Reply:
go up above - click on commits, it's the panel at the top of that page

Reply:
I see - I don't know what it looked like previously. I wonder if it just because I'm asking to merge from my downstream repo? In that case, the commit wouldn't belong to your repo (i think)

I think it is likely because I used the force push to otherwise clean things up. 

Let's try to merge it like normal, and then if there are any issues, I'll fall back on the strategy above (closing this PR and staring "fresh")